### ISSUE TITLE
Fix bug in `pow.rs`

### ISSUE THREAD

**Issue by tcharding** [rust-bitcoin/rust-bitcoin#2835]
There is a bug in `pow`  shown by this diff

```diff
--- a/bitcoin/src/pow.rs
+++ b/bitcoin/src/pow.rs
@@ -887,7 +887,7 @@ impl U256 {
         // (This is why we only care if the other non-msb dropped bits are all 0 or not,
         // so we can just OR them to make sure any bits show up somewhere.)
         let mantissa =
-            (mantissa + ((dropped_bits - (dropped_bits >> 127 & !mantissa)) >> 127)) as u64;
+            (mantissa + ((dropped_bits + (dropped_bits >> 127 & !mantissa)) >> 127)) as u64;
         // Step 6: Calculate the exponent
         // If self is 0, exponent should be 0 (special meaning) and mantissa will end up 0 too
         // Otherwise, (255 - n) + 1022 so it simplifies to 1277 - n
```

Fix it by doing:

- Write a unit test that fails on current master (put it as patch 2)
- Add the diff above as patch 1

**Comment by tcharding** [comment]
Might be interesting for you @jamillambert

**Comment by tcharding** [comment]
Props to @brunoerg for finding the bug in #2829.

**Comment by Kixunil** [comment]
Closed in #2834 (and #2837).
